src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,582 @@
+      var validationFlag = false
+      if (daysBetween == 0) {
+        validationFlag = Helpers.pathExists(s"${auditlogprefix_source_aws}/date=${fromDT.toString}")
+      } else {
+        val presentPaths = datesStream(fromDT).takeWhile(_.isBefore(untilDT)).toArray
+          .map(dt => s"${auditlogprefix_source_aws}/date=${dt}")
+          .filter(Helpers.pathExists)
+        if (presentPaths.length == daysBetween) {
+          validationFlag = true
+        }
+      }
+      if (validationFlag) {
+        DeploymentValidationReport(true,
+          getSimpleMsg("Validate_AuditLogPrefix"),
+          s"""Folder should be present from :${fromDT} to:${untilDT} """,
+          Some("SUCCESS"),
+          Some(workspace_id)
+        )
+      } else {
+        val msg =
+          s"""ReValidate the folder existence
+             | Make sure folder with required date folder exist inside ${auditlogprefix_source_aws}/workspaceId=${workspace_id}
+             |, primordial_date:${primordial_date}
+             |, maxDate:${maxDate} """.stripMargin
+        logger.log(Level.ERROR, msg)
+        DeploymentValidationReport(false,
+          getSimpleMsg("Validate_AuditLogPrefix"),
+          s"""Folder should be present from :${fromDT} to:${untilDT} """,
+          Some(msg),
+          Some(workspace_id)
+        )
+      }
+
+    } catch {
+      case exception: Exception =>
+        val msg =
+          s"""AuditLogPrefixTest workspace_id:${workspace_id}
+             | Make sure folder with required date folder exist inside ${auditlogprefix_source_aws}
+             |, primordial_date:${primordial_date}
+             |, maxDate:${maxDate} """.stripMargin
+        logger.log(Level.ERROR, msg)
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        DeploymentValidationReport(false,
+          getSimpleMsg("Validate_AuditLogPrefix"),
+          "Validating folders",
+          Some(fullMsg),
+          Some(workspace_id)
+        )
+    }
+  }
+
+  /**
+   * Performs validation of event hub data for Azure
+   *
+   * @param workspace_id
+   * @param scope
+   * @param key
+   * @param ehName
+   */
+  private def validateEventHub(workspace_id: String, scope: String, key: String, ehName: String, outputPath: String): DeploymentValidationReport = {
+    val testDetails = s"""Connectivity test with ehName:${ehName} scope:${scope} SecretKey_DBPAT:${key}"""
+    try {
+      import org.apache.spark.eventhubs.{ConnectionStringBuilder, EventHubsConf, EventPosition}
+      val ehConn = dbutils.secrets.get(scope = scope, key)
+      //TODO limit 10 check aws
+      //pull 10 milit from aws audit log and grab org id
+      val connectionString = ConnectionStringBuilder(
+        PipelineFunctions.parseAndValidateEHConnectionString(ehConn, false))
+        .setEventHubName(ehName)
+        .build
+
+      val ehConf = EventHubsConf(connectionString)
+        .setMaxEventsPerTrigger(100)
```

Comment:
I see that this is only called from a manual validation step (not every pipeline run), Love that. Let's go ahead and set this to 5000 to get more data to verify a broader set of data from more services / actions within the dataset.

Reply:
done